### PR TITLE
Add support for Phone mention without SMS domain `@expensify.sms`

### PR DESCRIPTION
**Pull request by FitseTLT**
### Fixed Issues
$ [GH_LINK](https://github.com/Expensify/App/issues/33902)

# Tests
1. What unit/integration tests cover your change? What autoQA tests cover your change?
1. What tests did you perform that validates your changed worked?
****
1. First on Expensify App change expensify-common dependency to point to latest commit of the current PR
https://github.com/Expensify/App/blob/f438cb7901c4405e7ec2bbb22aaa904bb5b8fb66/package.json#L103-L104
```
    "expensify-common": "git+ssh://git@github.com/Expensify/expensify-common.git#562c315123a44c4386b85eb02c5b90b20db901fa",

```
2. run `npm i`
3. Change suggestion code in order to test so change 
https://github.com/Expensify/App/blob/f438cb7901c4405e7ec2bbb22aaa904bb5b8fb66/src/pages/home/report/ReportActionCompose/SuggestionMention.js#L166
```
                    login: Str.removeSMSDomain(detail.login),

```

https://github.com/Expensify/expensify-common/assets/38649957/83961ea1-03fd-4f6f-9e4a-5beb626b0d13

# QA
1. What does QA need to do to validate your changes?
1. What areas to they need to test for regressions?
***
same

### CODE REVIEW

**Comment by FitseTLT** [comment]
@srikarparsi Don't worry about it. @c3024 and @jasperhuangg will handle the review

File: index.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+// const replacement = (match, g1, g2) => {
+//     console.log({ match, g1, g2 });
+//     // if (!Str.isValidMention(match)) {
+//     //     return match;
+//     // }
+//     return `${g1}<mention-user>${g2}</mention-user>`;
+// };
+const parser = new ExpensiMark();
+console.log(parser.replace(`sdflkj! @251-1236-09879878989.kl`));
```
Comment (jasperhuangg) [inline index.js:19]: remove this

**Review by jasperhuangg — CHANGES_REQUESTED** [review]

**Comment by c3024** [comment]
This appears to be working fine but I found a couple of issues that do not appear to be related this.

Screenshot 2024-02-14 at 6 57 17 PM


https://github.com/Expensify/expensify-common/assets/102477862/e2ed7c07-1430-4842-8b7a-74ef9dd1fa74



1. For users (email ids and phone numbers) with whom the sender has not interacted, they can send email ids as mentions but backend returns an error if they send a phone number as a mention.


https://github.com/Expensify/expensify-common/assets/102477862/e6440b45-8eeb-4856-8512-dff3e67b1989



2. For some numbers if I try to start a chat directly it says they are VoIP or Landline numbers. But if I send these numbers as mentions and click on them then a chat can be started.

Should they be fixed before proceeding with this? @jasperhuangg

**Comment by c3024** [comment]
Could you add unit tests for these changes in `_tests_`? @FitseTLT

**Comment by FitseTLT** [comment]
> Could you add unit tests for these changes in `_tests_`? @FitseTLT

It was more proper to update the existing test, so updated!

**Comment by c3024** [comment]
Please refer to [this](https://github.com/Expensify/expensify-common/pull/647#issuecomment-1943792379) as well. @FitseTLT

**Comment by FitseTLT** [comment]
@c3024 Yes You are right I did reproduce the error too but obviously it means there is a BE job also required in parallel to these changes of removing the sms domain in phone mentions.
Now if you mention a phone number contact that you haven't chatted before BE returns error
```
{
  "2629349569912594160": {
    "errors": {
      "1707933996120892": "Auth CreateReportAction returned an error"
    }
  }
}
```
cc: @jasperhuangg

**Comment by jasperhuangg** [comment]
Cool, I'll take a look at fixing that

**Comment by jasperhuangg** [comment]
@FitseTLT I investigated a back-end solution, and found that the proper solution should actually lie in the front-end.

[Here](https://github.com/Expensify/expensify-common/blob/45d3b61bb38b4f9a19ddf573ce1e212369b242db/lib/ExpensiMark.js#L165-L173), when we see that the mentioned user is a phone number (that doesn't have `@expensify.sms`), we should append `@expensify.sms` to the comment. This is because the back-end assumes that the text inside of the `<mention-user>` should be a valid user login. For SMS accounts, logins have to end with `@expensify.sms`.

Then, when we render the comment, we should omit the `@expensify.sms` from the comment and the user's account page.

Let me know if you have any additional questions, and sorry I didn't catch this sooner!

**Comment by jasperhuangg** [comment]
@FitseTLT I'm going to convert this to a draft while you work on it, let me know when it's ready for another review!

**Comment by FitseTLT** [comment]
@c3024 @jasperhuangg Ready for review

**Comment by c3024** [comment]
Did you check if the changes you made are working correctly for the issues discussed in the previous comments till date? @FitseTLT

**Comment by FitseTLT** [comment]
> Did you check if the changes you made are working correctly for the issues discussed in the previous comments till date? @FitseTLT

Yep

**Comment by c3024** [comment]
Could you post a video of a user starting a chat with a new number?

**Comment by FitseTLT** [comment]
https://github.com/Expensify/expensify-common/assets/38649957/17c19db5-a3a8-49d6-ba67-c41d5d6c52f0

**Comment by c3024** [comment]
From @jasperhuangg 's [comment here](https://github.com/Expensify/expensify-common/pull/647#issuecomment-1955248478)

> Then, when we render the comment, we should omit the `@expensify.sms` from the comment and the user's account page.
> 
> Let me know if you have any additional questions, and sorry I didn't catch this sooner!

@FitseTLT Can you see if you can omit the `@expensify.sms` from the report action, LHN subtitle and user's account page?

For report action this suggestion by @aim-salam in an earlier proposal for this issue appears to be useful.

```
props.tnode.init.textNode.data = props.tnode.init.textNode.data.split('@expensify')[0]
```
(`formatPhoneNumber` or `removeSMSDomain` is preferred instead of splitting like this)

for `MentionUserRenderer.tsx` [here](https://github.com/Expensify/App/blob/440c43eaf36c4844978c57ed5202a23f66efc290/src/components/HTMLEngineProvider/HTMLRenderers/MentionUserRenderer.tsx#L85)

**Comment by FitseTLT** [comment]
> From @jasperhuangg 's [comment here](https://github.com/Expensify/expensify-common/pull/647#issuecomment-1955248478)
> 
> > Then, when we render the comment, we should omit the `@expensify.sms` from the comment and the user's account page.
> > Let me know if you have any additional questions, and sorry I didn't catch this sooner!
> 
> @FitseTLT Can you see if you can omit the `@expensify.sms` from the report action, LHN subtitle and user's account page?
> 
> For report action this suggestion by @aim-salam in an earlier proposal for this issue appears to be useful.
> 
> ```
> props.tnode.init.textNode.data = props.tnode.init.textNode.data.split('@expensify')[0]
> ```
> 
> (`formatPhoneNumber` or `removeSMSDomain` is preferred instead of splitting like this)
> 
> for `MentionUserRenderer.tsx` [here](https://github.com/Expensify/App/blob/440c43eaf36c4844978c57ed5202a23f66efc290/src/components/HTMLEngineProvider/HTMLRenderers/MentionUserRenderer.tsx#L85)

Yeah definitely! We we will deal with the removal of the sms domain on the main issue only focusing on the phone number mention parsing here 👍

**Review by jasperhuangg — APPROVED** [review]

**Pull request merged**